### PR TITLE
Fix ICustomHitTest

### LINKED ISSUE #5879: ICustomHitTest not working
**Describe the bug**
It seems that implementing `ICustomHitTest` on `Shape` doesn't have any effect. It _does_ work with the immediate renderer `UseDeferredRendering=false`) though.

**To Reproduce**

[shape_custom_hittesting_bug.patch.txt](https://github.com/AvaloniaUI/Avalonia/files/6434227/shape_custom_hittesting_bug.patch.txt)

**Expected behavior**
I would expect the custom hit testing to work.

**Desktop (please complete the following information):**

- OS: Windows 10
- Version: master

### PR DESCRIPTION
**Pull request by mat1jaczyyy**
## What does the pull request do?

ICustomHitTest wasn't being handled correctly when using deferred renderer. Immediate renderer functions as intended.

## What is the current behavior?

Deferred renderer passes incorrectly translated points for clipping, and for actual hit testing doesn't test at all.

## What is the updated/expected behavior with this PR?

A repro is provided in #5879. Deferred renderer now passes non-translated points and does actual hit testing. On an slightly less related note, upon inspection it's possible to expand the hit window using this interface, so I've removed an old comment which claimed it was not possible. This is intended behavior.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [x] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes

People using deferred renderer will now have non-translated points passed to `HitTest`.

## Fixed issues

Fixes #5879

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5923) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5923) before we can accept your contribution.<br/><br/>:x: mat1jaczyyy [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5923)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=5923) it.</sub>

**Review by MarchingCube — APPROVED** [review]
LGTM.

File: src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs [hunk 1]
```diff
@@ -311,8 +312,7 @@ private bool FilterAndClip(IVisualNode node, ref Rect? clip)
 
                     if (!clipped && node.Visual is ICustomHitTest custom)
```
Comment (wieslawsoltes) [inline src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs:313]: What about ICustomSimpleHitTest ?

Comment (mat1jaczyyy) [reply src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs:313]: nice catch

**Pull request merged**